sdk/nodejs/cloudwatch/metric.ts

```diff
@@ -0,0 +1,402 @@
+        result = hasOwnProperty(change, "statistic") ? result.withStatistic(change.statistic) : result;
+        result = hasOwnProperty(change, "extendedStatistic") ? result.withExtendedStatistic(change.extendedStatistic) : result;
+        result = hasOwnProperty(change, "unit") ? result.withUnit(change.unit) : result;
+        return result;
+    }
+
+    public withDimensions(dimensions: pulumi.Input<Record<string, any>> | undefined) {
+        return new Metric({
+            ...this,
+            dimensions,
+        }, this.resource);
+    }
+
+    public withPeriod(period: pulumi.Input<number> | undefined) {
+        return new Metric({
+            ...this,
+            period,
+        }, this.resource);
+    }
+
+    public withUnit(unit: pulumi.Input<MetricUnit> | undefined) {
+        return new Metric({
+            ...this,
+            unit,
+        }, this.resource);
+    }
+
+    public withStatistic(statistic: pulumi.Input<MetricStatistic> | undefined) {
+        // If they're supplying a statistic, then we want to clear out extendedStatistic.
+        return new Metric({
+            ...this,
+            statistic,
+            extendedStatistic: pulumi.output([statistic, this.extendedStatistic])
+                                     .apply(([statistic, extendedStatistic]) =>
+                                        statistic !== undefined ? undefined : extendedStatistic),
+        }, this.resource);
+    }
+
+    public withExtendedStatistic(extendedStatistic: pulumi.Input<number> | undefined) {
+        // If they're supplying an extendedStatistic, then we want to clear out statistic.
+        return new Metric({
+            ...this,
+            statistic: pulumi.output([this.statistic, extendedStatistic])
+                             .apply(([statistic, extendedStatistic]) =>
+                                extendedStatistic !== undefined ? undefined : statistic),
+            extendedStatistic,
+        }, this.resource);
+    }
+
+    public createAlarm(name: string, args: AlarmArgs, opts: pulumi.CustomResourceOptions = {}) {
+        const comparisonOperator = utils.ifUndefined(args.comparisonOperator, "GreaterThanOrEqualToThreshold");
+        return new alarm.MetricAlarm(name, {
+            ...args,
+
+            comparisonOperator,
+            actionsEnabled: utils.ifUndefined(args.actionsEnabled, true),
+            alarmDescription: computeDescription(this, args, comparisonOperator),
+            treatMissingData: utils.ifUndefined(args.treatMissingData, "missing"),
+
+            dimensions: this.dimensions,
+            extendedStatistic: this.extendedStatistic.apply(s => s === undefined ? undefined : `p${s}`),
+            metricName: this.name,
+            namespace: this.namespace,
+            period: this.period,
+            statistic: this.statistic,
+            unit: this.unit,
+        }, { parent: this.resource, ...opts });
+    }
```

Comment:
needs docs, but this makes it easy to create a real cloudwatch alarm from a given metric.